### PR TITLE
[BUGFIX] Metric table.column_type should properly evaluate for snowflake

### PR DESCRIPTION
**Pull request by TrangPham**
- [ ] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [ ] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [ ] Code is linted - run `invoke lint` (uses `ruff format` + `ruff check`)
- [ ] Appropriate tests and docs have been updated

For more information about contributing, visit our [community resources](https://docs.greatexpectations.io/docs/core/introduction/community_resources#contribute-code-or-documentation).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

File: tests/integration/data_sources_and_expectations/expectations/test_expect_column_values_to_be_in_type_list.py [hunk 1]
```diff
@@ -135,3 +152,178 @@ def test_failure(
...
+        # pytest.param(
+        #     gxe.ExpectColumnValuesToBeInTypeList(column="ARRAY", type_list=["ARRAY"]),
+        #     id="ARRAY",
+        # ),
+    ],
+)
+@parameterize_batch_for_data_sources(
+    data_source_configs=[
+        SnowflakeDatasourceTestConfig(
+            column_types={
```
Comment (Shinnnyshinshin) [inline tests/integration/data_sources_and_expectations/expectations/test_expect_column_values_to_be_in_type_list.py:262]: 🚀 straight from the CBT. Love it

File: tests/integration/data_sources_and_expectations/expectations/test_expect_column_values_to_be_in_type_list.py [hunk 2]
```diff
@@ -135,3 +152,178 @@ def test_failure(
...
+                "2021-01-03 00:00:00",
+            ],
+            "TIMESTAMP_TZ": [
+                "2021-01-01 00:00:00",
+                "2021-01-02 00:00:00",
+                "2021-01-03 00:00:00",
+            ],
+            "TINYINT": [1, 2, 3],
+            "VARBINARY": [b"1", b"2", b"3"],
+            # Complex data types which are not hashable by testing framework currently
```
Comment (Shinnnyshinshin) [inline tests/integration/data_sources_and_expectations/expectations/test_expect_column_values_to_be_in_type_list.py:311]: Super helpful comment. Thank you 🙇🏼

**Review by Shinnnyshinshin — APPROVED** [review]
LGTM!

File: great_expectations/expectations/metrics/util.py [hunk 3]
```diff
@@ -415,10 +415,14 @@ def get_sqlalchemy_column_metadata(
 
         dialect_name = execution_engine.dialect.name
         if dialect_name == GXSqlDialect.SNOWFLAKE:
```
Comment (TrangPham) [inline great_expectations/expectations/metrics/util.py:417]: If in ...

File: great_expectations/expectations/metrics/util.py [hunk 4]
```diff
@@ -415,10 +415,14 @@ def get_sqlalchemy_column_metadata(
...
+            # WARNING: Do not alter columns in place, as they are cached on the inspector
+            columns_copy = [column.copy() for column in columns]
+            for column in columns_copy:
+                column["type"] = column["type"].compile(dialect=execution_engine.dialect)
             return [
```
Comment (TrangPham) [inline great_expectations/expectations/metrics/util.py:422]: if snowflake: return [...] else return columns_copy

File: great_expectations/expectations/core/expect_column_values_to_be_of_type.py [hunk 5]
```diff
@@ -411,6 +411,15 @@ def _validate_sqlalchemy(self, actual_column_type, expected_type, execution_engi
 
         if expected_type is None:
             success = True
+        elif execution_engine.dialect_name == GXSqlDialect.SNOWFLAKE:
```
Comment (TrangPham) [inline great_expectations/expectations/core/expect_column_values_to_be_of_type.py:414]: if in

File: great_expectations/expectations/core/expect_column_values_to_be_in_type_list.py [hunk 6]
```diff
@@ -454,6 +457,15 @@ def _validate_pandas(  # noqa: C901, PLR0912
     def _validate_sqlalchemy(self, actual_column_type, expected_types_list, execution_engine):
         if expected_types_list is None:
             success = True
+        elif execution_engine.dialect_name == GXSqlDialect.SNOWFLAKE:
```
Comment (TrangPham) [inline great_expectations/expectations/core/expect_column_values_to_be_in_type_list.py:460]: if in

**Pull request merged**